torch/nn/modules/module.py

```diff
@@ -524,17 +524,17 @@ def _slow_forward(self, *input, **kwargs):
 
     def __call__(self, *input, **kwargs):
         for hook in self._forward_pre_hooks.values():
-            hook(self, input)
+            result = hook(self, input)
+            if result:
```

Comment:
This seems wrong? We either need to clearly document that you're supposed to return a tuple from this, or we have to auto-wrap the input in a tuple if a single value has been returned from the hook. Think about the case when someone returns a single tensor.

Reply:
sure, I'll wrap the input

Reply:
will there be a case when the forward function accepts a tuple as input?

Reply:
Yes, this is very much possible. If we want to go down the wrapping path, then I'd create a singleton tuple if `not isinstance(result, tuple)`.

Reply:
So if the forward function accepts a tuple as input, should we wrap the tuple again in another singleton tuple?

Reply:
Well what we're doing is a bit magical and in general hard to do, but I think it's also pretty consistent and user-friendly. We can document the edge cases, but I don't think they will be very common.